Modules, Exports, and Imports.md

```diff
@@ -0,0 +1,301 @@
+"blacklist" private items). A list of items is separated by commas, and may be broken up across many lines. The one
+wrinkle is how union types are exported, but that will be addressed below.
+
+```elm
+module MyModule (MyType, myValue, myFunction) where
+```
+
+If `MyModule` defines `myPrivateFunction`, it cannot be imported by any other module, regardless of syntax used.
+
+Although uncommon, you should avoid situations like this:
+
+```elm
+module Chance (Model, init) where
+
+type Coin = Heads | Tails
+
+type alias Model = { coin : Coin }
+
+init : Model
+init = Model Heads
+```
+
+The private `Coin` type is visible in the definition of the public `Model` type. Even worse, someone importing this code
+can actually obtain a value of type `Coin` through `init`, even though that type doesn't exist to them. And worst of
+all, the compiler will not catch this error. It *should*, but it doesn't. So be careful when defining types, and think
+about what will and will not be exported.
+
+## Imports
+The most common way to import a module is also the simplest:
+
+```elm
+import Dict
+```
+
+This gives you access to everything in the [Dict
+library](http://package.elm-lang.org/packages/elm-lang/core/latest/Dict) by prefacing it with the module name and a dot.
+This is sometimes called a **qualified** import. So if you want to use `Dict.insert`, this import statement is sufficient.
+
+Qualified imports are the preferred way to import values from other modules, because you can always tell where something
+came from by the module name right in front of it. Conversely, it makes it easy to tell what is defined in *this* module.
+
+The alternative to a qualified import is an **exposed** import, because of the keyword `exposing` (whose length should
+serve as a deterrent to using it!). But it's usually okay if you list what you're exposing, for example:
+
+```elm
+import Dict exposing (Dict)
+import Html exposing (div, span, h1, h2)
+```
+
+In the first case, the `Dict` in parentheses is a type, not the module. (It's very common for modules to define types of
+the same name as the module itself.) It avoids the need to say `Dict.Dict` in type annotations. In the second case, the
+functions being exposed have distinctive names, and are going to be used frequently.
+
+In both cases, the module name is also available to be use qualified. So you can still use `Dict.insert` or `Html.text`.
+When you have a list of exposed imports, the order doesn't matter, but typically the types are listed before the values.
+
+The real reason exposed imports are considered an antipattern is because they can be combined with `(..)` to dump the
+entire module into the current one. When there are multiple such imports, it becomes impossible to see where something
+came from, and the odds of a name collision is much higher.
+
+```elm
+-- Antipatten!
+import Html exposing (..)
```

Comment:
This is one is very common, and Html exposes usual functions you want to use so I'm not sure if you should call this an anti pattern.  Maybe `import Effects exposing (...)` is a better one. Because Effects has`map` and Signal also has `map`